src/OCPNPlatform.cpp

```diff
@@ -1302,11 +1312,94 @@ wxString &OCPNPlatform::GetPrivateDataDir()
         m_PrivateDataDir = androidGetPrivateDir();
 #endif
     }
-    
     return m_PrivateDataDir;
 }
 
 
+/** Expand environment and drop non-existing paths in input.*/
+static wxString normalizeLoadPath(const wxString dirs) {
+    wxString loadPath("");
+    wxStringTokenizer tokens(dirs, PATH_SEP);
+    while (tokens.HasMoreTokens()) {
+        wxString dir = tokens.GetNextToken();
+        if (dir.EndsWith(wxFileName::GetPathSeparator())) {
+            dir.RemoveLast();
+        }
+        if (!dir.EndsWith("opencpn")) {
+            dir += wxFileName::GetPathSeparator();
+            dir += "opencpn";
+        }
+        wxFileName path(dir);
+        path.Normalize();
+        if (!path.Exists()) {
+            continue;
+        }
+        if (loadPath != "") {
+            loadPath += PATH_SEP;
+        }
+        loadPath += path.GetFullPath();
+    }
+    return loadPath;
+}
+
+
+wxString OCPNPlatform::GetPluginLoadPath()
+{
+    if (m_pluginLoadPath != "" ) {
+        return m_pluginLoadPath;
+    }
+    wxString dirs("");
+    if (wxPlatformInfo::Get().GetOperatingSystemId() & wxOS_UNIX_LINUX) {
+        if (!wxGetEnv("XDG_DATA_DIRS", &dirs)) {
+            dirs = LINUX_PLUGIN_LOAD_PATH;
+        }
+        dirs.replace(dirs.begin(), dirs.end(), ":", ";");
+    }
+    else if (wxPlatformInfo::Get().GetOperatingSystemId() & wxOS_WINDOWS) {
+        dirs = GetWinPluginBaseDir();
+    }
+    m_pluginLoadPath = NormalizePath(dirs);
+    if (m_pluginLoadPath != "") {
+        m_pluginLoadPath += PATH_SEP;
+    }
+    m_pluginLoadPath += GetPluginDir();
+    if (m_pluginLoadPath.EndsWith(wxFileName::GetPathSeparator())) {
+	m_pluginLoadPath.RemoveLast();
+    }
+    wxLogMessage("Using plugin load path: %", m_pluginLoadPath);
+    return m_pluginLoadPath;
+}
+
+
+wxString& OCPNPlatform::GetWinPluginBaseDir()
+{
+    if (g_winPluginDir != "") {
```

Comment:
Is it possible to do without the global variable?

Reply:
Anything is possible. But the whole application is built around globals, and specifically for things which is visible in the UI like this one it would create some very different code paths for a single variable.

I agree that there are too many globals. But changing this would IMHO require some kind of separate, overall  effort.

Unless you have any ideas?

Reply:
I dream of getting rid of globals, but you are right - probably not worth it here.

 I guess my issue is also trying to understand the difference between `OCPNPlatform::GetWinPluginBaseDir` and `g_winPluginDir`. The "accessor" modifying a global is weird - and I fear we will end up in some years with half the code using `g_winPluginDir` directly and half `GetWinPluginBaseDir`, with bugs depending on whether `GetWinPluginBaseDir` is called before a `g_winPluginDir` access or not.

Reply:
I have basically just tried to follow the existing code when it comes to the UI - and those parts relies on globals.

The accessor is basically geared towards the PluginHandler, which uses the value. I actually think very few users will change it from the default. But since Dave want it.

If I follow your line of reasoning we could hide the thing completely in a file with getter/setter. Perhaps it would be better, dunno... or wait... yes! 

will fix.

Reply:
You might be right. I'll look into a classic hidden variable with getter/setter, it it works with the UI code.